L1Trigger/L1CaloTrigger/interface/Phase2L1CaloEGammaUtils.h

```diff
@@ -486,6 +480,12 @@ namespace p2eg {
       };
     };
 
+    // overload operator= to use copy constructor
+    towers3x4 operator=(const towers3x4& other) {
+      const towers3x4& newTowers3x4(other);
+      return newTowers3x4;
+    };
```

Comment:
This copy assignment operator does not follow the usual semantics. It constructs a new `towers3x4` object from the argument, and returns that new object (or a copy of the new object if for any reason the compiler won't elide the copy), and leaves `this` object untouched. The behavior should be that the contents of the argument are copied to `this`, and `*this` is returned by reference.

How about leaving both copy assignment and copy constructor to be generated by the compiler?

(on a further note, independently of this PR, the names `towers3x4` and `towerHCAL` are against our naming convention, class names should start with capital letter; and the `towersHCAL` data member name should have a trailing underscore, i.e. be `towersHCAL_`).

Reply:
> > How about leaving both copy assignment and copy constructor to be generated by the compiler?
> 
> I'm not sure default copy-constructor/copy-assignment operator will correctly copy `towersHCAL` 2D array.

Ah right, C-arrays don't have copy operators 🤦 . Maybe the `towersHCAL` could be changed to be nested `std::array` (naively looks straightforward), after which the copy (and move) operations would be defined automatically?

> > This copy assignment operator does not follow the usual semantics. It constructs a new towers3x4 object from the argument, and returns that new object (or a copy of the new object if for any reason the compiler won't elide the copy), and leaves this object untouched. The behavior should be that the contents of the argument are copied to this, and *this is returned by reference.
> 
> Should we also change [`region3x4::operator=`](https://github.com/iarspider/cmssw/blob/fea46e71a12c3e54c78b51c93ab4975fd691c017/L1Trigger/L1CaloTrigger/interface/Phase2L1CaloEGammaUtils.h#L408)?

Yes, that should be fixed as well.